src/content/zh-CN/2020/performance.md

```diff
@@ -0,0 +1,396 @@
+  caption="按国家分类的 TTFB 总性能",
+  description="条形图显示 TTFB 的性能始终低于标准，在28个国家中，只有4个拥有超过30％的 TTFB 良好的网站。有大量的网站被归类为需要改善（始终高于40％），不良体验的比例随着排名的降低而上升。",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vT3ukFCTxx__dTBLwDQ3K_hCtXdLRGOGUYdR_TO043n_aCTdcwkxPUku9Qfoal6BESiV5RMtd0nEbqT/pubchart?oid=1135415956&format=interactive",
+  width="645",
+  height="792",
+  sheets_gid="1440255644",
+  sql_file="web_vitals_by_country.sql"
+  )
+}}
+
+今年的 TTFB 地理分布图和[2019年结果](../2019/performance#ttfb-by-geo)相似，都表明了有更多的站点变快了，但是和 FCP 相似，阈值都发生了改变。以前，我们认为 TTFB 在200ms以下为快，而在1000ms以上为慢。 2020年，低于500ms的TTFB良好，高于1500ms的TTFB较差。这些评级方面的许多变化可以解释我们观察到的重大改变，比如韩国有着良好用户体验的站点增加了36％，台湾增加了22%。总体上，我们仍然看到了一些和往年的相似点，例如亚太地区和某些欧洲地区仍然处于领先地位。
+
+### 按连接类型查看 TTFB
+
+{{ figure_markup(
+  image="performance-ttfb-by-connection-type.png",
+  caption="按连接类型分类的 TTFB 总性能", 
+  description="条形图显示，TTFB 受到连接类型的严重影响，在4G和离线的场景下，分别只有21％和22％的体验良好。其他连接类型几乎没有良好的 TTFB 体验（3G上为1％除外）。",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vT3ukFCTxx__dTBLwDQ3K_hCtXdLRGOGUYdR_TO043n_aCTdcwkxPUku9Qfoal6BESiV5RMtd0nEbqT/pubchart?oid=810992122&format=interactive",
+  sheets_gid="306222260",
+  sql_file="web_vitals_by_ect.sql"
+  )
+}}
+
+TTFB受网络延迟和连接类型的影响。由上图可知，延迟越高、连接越慢，TTFB度量值就会越差。即使在被认为快速（4G）的移动端连接上，也只有21%的网站有一个快速的 TTFB。低于4G速度的站点基本上不会被评定为快速。
+
+我们来看看[2018年12月-2019年11月全球移动端网速](https://www.speedtest.net/insights/blog/content/images/2020/02/Ookla_Mobile-Speeds-Poster_2020.png)，可以看出整体上移动端连接网速并不高。那些网速和代表蜂窝网络的技术（比如5G）分布不均，并且最终会影响 TTFB。举个例子，[我们看尼日利亚的网络分布图]，该国大部分区域覆盖的都是2G和3G，极少数区域才有4G网络分布。
+
+令人惊讶的是，离线站点和通过4G网络访问的站点有着相同数目的良好的TTFB。通过 service workers，我们可以预期一些 TTFB 问题会得到缓解，但该趋势并未在上图中反映出来。
+
+## 性能监测的使用
+
+有许多不同的以用户为中心的指标可用于评估网站和应用程序。然而，有时预定义的指标并不完全符合我们的特定场景和需要。[性能监测对象 API](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver)允许我们通过[自定义时间测量 API](https://developer.mozilla.org/en-US/docs/Web/API/User_Timing_API)、[Long Tasks API](https://developer.mozilla.org/en-US/docs/Web/API/Long_Tasks_API)、[Event Timing API](https://web.dev/custom-metrics/#event-timing-api)和[一些底层 APIs](https://web.dev/custom-metrics/)去获取自定义度量数据。举例来说，在这些API的帮助下我们可以记录页面转场时间或者量化服务端渲染应用的客户端激活。
+
+{{ figure_markup(
+  image="performance-performance-observer-usage.png",
+  caption="按设备分类的性能监测的使用",
+  description="条形图显示，性能监测 API 的采用率很低，桌面端为6.6％，移动端为7％。",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vT3ukFCTxx__dTBLwDQ3K_hCtXdLRGOGUYdR_TO043n_aCTdcwkxPUku9Qfoal6BESiV5RMtd0nEbqT/pubchart?oid=632678090&format=interactive",
+  sheets_gid="934401790",
+  sql_file="performance_observer.sql"
+  )
+}}
+
+上图显示，被跟踪的站点中有6-7％使用了性能监测，它具体取决于设备类型。这些站点借助这些底层 API 去创建自定义度量指标，并通过性能监测 API 去核对，然后可能通过其他性能报告工具去使用。这样的采用率表明人们更倾向于使用预定义的指标（比如 Lighthouse里面定义的指标），但那些特定应用场景的 API 也令人印象深刻。
+
+## 总结
+
+用户体验不仅是某一个特定范围的问题，还取决于多种因素。要试图了解性能状态而又不排除欠佳的用户体验，我们必须通盘考虑。每次访问网站都在讲述一个故事。我们的个人和国家的社会经济状况决定了我们可以负担得起的设备和互联网提供商的类型。我们居住地的地理位置影响了延迟（我们澳大利亚人经常能感受到这种痛苦），而经济状况决定了可用的蜂窝网络覆盖范围。我们在浏览什么网站？我们浏览网站的目的是什么？在构建所有人都可以快速访问的用户体验过程中，背景对于分析数据和发展必要的同理心和关怀都至关重要。
+
```

Comment:
```suggestion
```

Reply:
this line is the end of this fie, it seems it is no need to remove the empty line, since other files have too ?